cache/cache_rep.h

```diff
@@ -0,0 +1,53 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "rocksdb/cache.h"
+#include "rocksdb/slice.h"
+
+// [DO-NOT-COMMIT] [A header inclusion issue][A temporary hack]
+// Including the following two dependencies in cache_rep.h leads to a wall of compiler errors when including cache_rep.h in write_buffer_manager.h
+// It is suspected to be circular dependency issue
+// Therefore I commented these two dependencies out and redefine the const from these two dependencies in this file [DO-NOT-COMMIT] 
+// #include "table/block_based/block_based_table_reader.h"
+// #include "util/coding.h"
+
+namespace ROCKSDB_NAMESPACE {
+class CacheRep {
+ public:
+  // The key will be longer than keys for blocks in SST files so they won't
+  // conflict.
+
+  // [DO-NOT-COMMIT][A header inclusion issue]
+  // The following four lines (including the two commented lines) are for adapting to the [A temporary hack] mentioned above
+  // static const std::size_t kCacheKeyPrefixSize =
+  //     BlockBasedTable::kMaxCacheKeyPrefixSize + kMaxVarint64Length;
+  static const std::size_t kMaxVarint64Length = 10;
+  static const std::size_t kCacheKeyPrefixSize = kMaxVarint64Length * 4 + 1;
+
+  explicit CacheRep(std::shared_ptr<Cache> cache);
+  Slice GetNextCacheKey();
+
+  std::shared_ptr<Cache> cache_;
+  std::mutex cache_mutex_;
+  std::atomic<std::size_t> cache_allocated_size_;
+  std::uint64_t next_cache_key_id_ = 0;
```

Comment:
[minor_improvement]
I declared "std::uint64_t next_cache_key_id_ = 0;" before "cache_key_" for better readability